server/bin/pbench-verify-backup

```diff
@@ -0,0 +1,134 @@
+#!/usr/bin/env python3
+# -*- mode: python -*-
+
+import os
+import sys
+import glob
+import hashlib
+import shutil
+import signal
+import configtools
+
+# Correct the first path to be the "lib" directory to make the
+# pbench-backup-to-s3 script simple, allowing the imports from the
+# backups3 library to work cleanly.
+first_path = sys.path[0]
+assert first_path.endswith('/bin')
+lib_path = os.path.join(os.path.dirname(first_path), 'lib')
+sys.path[0] = lib_path
+
+from s3backup import s3Config, Entry
+from pbench import PbenchConfig, BadConfig
+
+archive_or_backup = {
+    "archive": "ARCHIVE: %s/%s: %s\n",
+    "backup": "BACKUP: %s/%s: %s\n"
+    }
+
+def checkmd5(target_dir, list_dir, indicator, report):
+    # Function to check integrity of results in a local (archive or local backup) directory
+    tarlist = glob.glob(os.path.join(target_dir, "*", "*.tar.xz"))
+    with open("%s/list.%s" % (list_dir, indicator), 'w') as f_list:
+        for tar in tarlist:
+            controller = tar.split("/")[-2]
+            result_name = tar.split("/")[-1]
+            md5 = ("%s.md5" % tar)
+            f_list.write("%s\n" % result_name)
+            with open(md5) as f:
+                md5_value = f.readline().split(" ")[0]
+            with open(tar, 'rb') as f:
+                data = f.read()
+                md5_returned = hashlib.md5(data).hexdigest()
+            if md5_value == md5_returned:
+                report.write(archive_or_backup[indicator] % (controller, result_name, "OK"))
+            elif md5 != md5_returned:
+                report.write(archive_or_backup[indicator] % (controller, result_name, "FAIL"))
+
+
+def compare_local_backup_with_s3_backup(pb_obj, s3_obj, Entry, list_dir, indicator, report):
+    # Function to check intergrity of results between local backup and s3
+    with open("%s/list.s3" % list_dir, 'w') as f_list:
+        s3_content_list = []
+        backup_content_list = []
+        kwargs = {'Bucket': s3_obj.bucket_name}
+        while True:
+            resp = s3_obj.s3_client.list_objects_v2(**kwargs)
+            for obj in resp['Contents']:
+                f_list.write("%s\n" % obj['Key'])
+                md5_returned = obj['ETag'].strip("\"")
+                s3_content_list.append(Entry(obj['Key'], md5_returned))
+                #s3_content_list.append({"name": obj['Key'], "md5sum": md5_returned})
+            try:
+                kwargs['ContinuationToken'] = resp['NextContinuationToken']
+            except expression as identifier:
+                break
+    with open("%s/list.%s" % (list_dir, indicator)) as f:
+        backup_content = f.readlines()
+    for content in backup_content:
+        result_name = content.strip("\n")
+        local_result_path = glob.glob(os.path.join(pb_obj.BACKUP, "*", result_name))
```

Comment:
If we need to trade space for time, we might factor out the glob call into its own function, and force the iterator to produce a list that we can then pass to the two functions that need it.